src/swtpm/swtpm.c

```diff
@@ -406,6 +406,9 @@ int swtpm_main(int argc, char **argv, const char *prgname, const char *iface)
         exit(EXIT_FAILURE);
     }
 
+    if (handle_log_options(logdata) < 0)
+        exit(EXIT_FAILURE);
+
```

Comment:
This has now been moved before this part here, which could now lead to permission issues when accessing existing log files.
```
    /* change process ownership before accessing files */
    if (runas) {
        if (change_process_owner(runas) < 0)
            goto exit_failure;
    }
```

Reply:
Thanks, I missed that. I'll move `change_process_owner()` before `handle_log_options()`.

Reply:
Tested this and it looks good.